src/ophyd_epics_devices/panda.py

```diff
@@ -92,11 +92,37 @@ class PVIEntry(TypedDict, total=False):
     x: str
 
 
-def block_name_number(block_name: str) -> Tuple[str, int]:
-    m = re.match("^([a-z]+)([0-9]*)$", block_name)
-    assert m, f"Expected '<block_name><block_num>', got '{block_name}'"
-    name, num = m.groups()
-    return name, int(num or 1)
+def block_name_number(block_name: str) -> Tuple[str, Optional[int]]:
+    """Maps a panda block name to a block and number.
+
+    There are exceptions to this rule; some blocks like pcap do not contain numbers.
+    Other blocks may contain numbers and letters, but no numbers at the end.
+
+    Such block names will only return the block name, and not a number.
+
+    If this function returns both a block name and number, it should be instantiated
+    into a device vector."""
+    m = re.match("^([0-9a-z_-]*)([0-9]+)$", block_name)
+    if m is not None:
+        name, num = m.groups()
+        return name, int(num or 1)  # just to pass type checks.
+
+    return block_name, None
+
+
+def _remove_inconsistent_blocks(pvi: Dict[str, PVIEntry]) -> None:
```

Comment:
Yeah seems like a reasonable thing to do for now until the root cause of this problem is found